Gifski/EditVideoViewController.swift

```diff
@@ -318,10 +343,41 @@ final class EditVideoViewController: NSViewController {
 			}
 		}()
 		let frameCount = duration * frameRateSlider.doubleValue
-		let dimensions = resizableDimensions.changed(dimensionsType: .pixels).currentDimensions.value
-		var fileSize = (Double(dimensions.width) * Double(dimensions.height) * frameCount) / 3
-		fileSize = fileSize * (qualitySlider.doubleValue + 1.5) / 2.5
-		estimatedSizeLabel.stringValue = "Estimated size: " + formatter.string(fromByteCount: Int64(fileSize))
+
+		estimatedSizeLabel.textColor = NSColor(named: "gray")
+		estimatedSizeLabel.stringValue = "..."
+		estimateWorkItem = DispatchWorkItem {
+			print("Total: \(frameCount)")
+
+			Gifski.setOnNewFrame { framesWritten in
+				if self.dataSize == 0 {
+					return
+				}
+
+				self.framesWritten = framesWritten
+				DispatchQueue.main.sync {
+					self.estimatedSizeLabel.stringValue = "Estimated size: " + self.formatter.string(fromByteCount: Int64(self.dataSize * (Int64(frameCount) / Int64(framesWritten))))
+				}
+			}
+			Gifski.setWriteCallback { size in
+				self.dataSize += Int64(size)
+			}
+			Gifski.run(conversion) { result in
+				do {
+					let data = try result.get()
+					self.gifData = data
+					self.estimatedSizeLabel.textColor = NSColor(named: "green")
+					self.estimatedSizeLabel.stringValue = "File size: " + self.formatter.string(fromByteCount: Int64(data.count))
+				} catch {
+					self.presentError(error, modalFor: self.view.window)
+				}
+			}
+		}
+		guard let workItem = estimateWorkItem else {
+			// This should never happen as we define estimateWorkItem above
```

Comment:
Would be good to use `assertionFailure()` just in case it actually happens. The comment could be the message.